apps/tup-cms/Dockerfile

```diff
@@ -1,14 +1,25 @@
-# TACC/Core-CMS#v4.4.2
-FROM taccwma/core-cms:01206e1
+FROM node:20 as node_build
+COPY package.json package-lock.json /code/
+WORKDIR /code
+RUN npm ci
+
+COPY . /code/
+RUN npx nx build tup-ui
+RUN npx nx build tup-cms-react
```

Comment:
**Question.** Why do you opt for `npx nx build …` instead of `npm run build …`? The latter is a familiar and common abstraction. If our build system changes, and the Dockerfile used `npm run …`, then the Dockerfile would not need to be updated.

Reply:
`npm run build` in an nx project only builds the "default" project (which would be tup-ui in our case IIRC). Since we also want to bundle the cms-react components, we have to use Nx here to specify the apps we want.

If we ever decide to drop Nx, we'll have to rearchitect the repo anyway, so updating the dockerfile would be the least of our worries.